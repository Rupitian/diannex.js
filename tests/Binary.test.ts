import * as path from "path";

import { Binary } from "../";

describe("Binary", () => {
    test("reads and processes v2 binary file correctly", () => {
        expect(() => {
            Binary.readFromFile(path.resolve(__dirname, "./files/out/general-v2.dxb"));
        }).toThrowError("Binary not for this version of Diannex.");
    });

    test("reads and processes v3 binary file correctly", () => {
        const bin: Binary = Binary.readFromFile(path.resolve(__dirname, "./files/out/sample.dxb"));

        expect(bin).toMatchSnapshot();
    });
});