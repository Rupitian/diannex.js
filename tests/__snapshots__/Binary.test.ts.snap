// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Binary reads and processes v3 binary file correctly 1`] = `
Binary {
  "definitions": Array [
    Object {
      "instructionIndex": -1,
      "reference": 0,
      "symbol": 4,
    },
    Object {
      "instructionIndex": -1,
      "reference": 1,
      "symbol": 5,
    },
    Object {
      "instructionIndex": -1,
      "reference": 2,
      "symbol": 6,
    },
    Object {
      "instructionIndex": 0,
      "reference": 3,
      "symbol": 7,
    },
  ],
  "externalFunctionList": Array [
    8,
    9,
    10,
    11,
    12,
    13,
  ],
  "functions": Array [
    Object {
      "instructionIndices": Array [
        323,
      ],
      "symbol": 3,
    },
  ],
  "instructions": Object {
    "data": Array [
      70,
      8,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      67,
      20,
      0,
      0,
      0,
      0,
      70,
      9,
      0,
      0,
      0,
      1,
      0,
      0,
      0,
      29,
      18,
      4,
      0,
      0,
      0,
      78,
      18,
      5,
      0,
      0,
      0,
      78,
      71,
      18,
      6,
      0,
      0,
      0,
      78,
      18,
      7,
      0,
      0,
      0,
      16,
      1,
      0,
      0,
      0,
      72,
      16,
      0,
      0,
      0,
      18,
      8,
      0,
      0,
      0,
      16,
      1,
      0,
      0,
      0,
      72,
      27,
      0,
      0,
      0,
      74,
      18,
      9,
      0,
      0,
      0,
      78,
      16,
      1,
      0,
      0,
      0,
      70,
      10,
      0,
      0,
      0,
      1,
      0,
      0,
      0,
      29,
      64,
      26,
      0,
      0,
      0,
      18,
      10,
      0,
      0,
      0,
      78,
      16,
      5,
      0,
      0,
      0,
      16,
      1,
      0,
      0,
      0,
      70,
      11,
      0,
      0,
      0,
      2,
      0,
      0,
      0,
      29,
      70,
      8,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      19,
      11,
      0,
      0,
      0,
      1,
      0,
      0,
      0,
      78,
      18,
      12,
      0,
      0,
      0,
      78,
      20,
      1,
      0,
      0,
      0,
      70,
      12,
      0,
      0,
      0,
      1,
      0,
      0,
      0,
      16,
      1,
      0,
      0,
      0,
      48,
      66,
      11,
      0,
      0,
      0,
      18,
      13,
      0,
      0,
      0,
      78,
      64,
      20,
      0,
      0,
      0,
      16,
      1,
      0,
      0,
      0,
      20,
      1,
      0,
      0,
      0,
      70,
      13,
      0,
      0,
      0,
      2,
      0,
      0,
      0,
      29,
      18,
      14,
      0,
      0,
      0,
      78,
      16,
      0,
      0,
      0,
      0,
      26,
      0,
      0,
      0,
      0,
      28,
      0,
      0,
      0,
      0,
      16,
      5,
      0,
      0,
      0,
      50,
      66,
      36,
      0,
      0,
      0,
      28,
      0,
      0,
      0,
      0,
      69,
      0,
      0,
      0,
      0,
      1,
      0,
      0,
      0,
      29,
      28,
      0,
      0,
      0,
      0,
      16,
      1,
      0,
      0,
      0,
      32,
      26,
      0,
      0,
      0,
      0,
      64,
      204,
      255,
      255,
      255,
      10,
      0,
      0,
      0,
      0,
      18,
      15,
      0,
      0,
      0,
      78,
      16,
      5,
      0,
      0,
      0,
      30,
      16,
      0,
      0,
      0,
      0,
      49,
      66,
      17,
      0,
      0,
      0,
      18,
      16,
      0,
      0,
      0,
      78,
      16,
      1,
      0,
      0,
      0,
      33,
      64,
      227,
      255,
      255,
      255,
      29,
      67,
      16,
      1,
      0,
      0,
      0,
      75,
      11,
      0,
      0,
      0,
      16,
      1,
      0,
      0,
      0,
      75,
      21,
      0,
      0,
      0,
      77,
      28,
      0,
      0,
      0,
      0,
      19,
      17,
      0,
      0,
      0,
      1,
      0,
      0,
      0,
      78,
      64,
      15,
      0,
      0,
      0,
      28,
      0,
      0,
      0,
      0,
      19,
      18,
      0,
      0,
      0,
      1,
      0,
      0,
      0,
      78,
      10,
      0,
      0,
      0,
      0,
      67,
    ],
    "type": "Buffer",
  },
  "scenes": Array [
    Object {
      "instructionIndices": Array [
        10,
      ],
      "symbol": 2,
    },
  ],
  "stringTable": Array [
    "narrator",
    "sample",
    "area0.intro",
    "area0.example",
    "menu.button_start",
    "menu.button_continue",
    "menu.button_quit",
    "menu.label_welcome",
    "getPlayerName",
    "char",
    "awardPoints",
    "deductPoints",
    "getFlag",
    "setFlag",
  ],
  "translationLoaded": true,
  "translationTable": Array [
    "Start",
    "Continue",
    "Quit",
    "Welcome, \${0}!",
    "Welcome to the test introduction scene!",
    "One quick thing I have to ask before you begin...",
    "Is this a question?",
    "Yes",
    "No",
    "That is correct.",
    "Hm... I don't believe that's correct.",
    "Either way, it was nice meeting you, \${0}.",
    "This is the end of the sample intro scene!",
    "By the way, I feel like I've said this before... quite strange.",
    "Well, now it's time for a loop!",
    "Or, a simpler loop!",
    "The same thing, over and over...",
    "This is an example function, being passed \${0}",
    "This is an example function (but an alternate line with 50% chance), being passed \${0}",
  ],
}
`;
